nixos/modules/services/misc/gogs.nix

```diff
@@ -164,13 +178,19 @@ in
       wantedBy = [ "multi-user.target" ];
       path = [ pkgs.gogs.bin ];
 
-      preStart = ''
+      preStart = let
+        runConfig = "${cfg.stateDir}/custom/conf/app.ini";
+      in ''
         # copy custom configuration and generate a random secret key if needed
         ${optionalString (cfg.useWizard == false) ''
           mkdir -p ${cfg.stateDir}/custom/conf
-          cp -f ${configFile} ${cfg.stateDir}/custom/conf/app.ini
-          KEY=$(head -c 16 /dev/urandom | tr -dc A-Za-z0-9)
-          sed -i "s,#secretkey#,$KEY,g" ${cfg.stateDir}/custom/conf/app.ini
+          cp -f ${configFile} ${runConfig}
+          KEY=$(head -c 16 /dev/urandom | base64)
```

Comment:
@Mic92: I think we are OK from the NixOS module side. I have changed it so that `SECRET_KEY` is only generated on the first start. After checking the Gogs code I have found that `SECRET_KEY` is used for 2-factor auth and CSRF token generation. Session cookies are generated from user password + random.